packages/nodes-base/nodes/If/test/v2/IfV2.node.test.ts

```diff
@@ -1,5 +1,155 @@
+import { mock } from 'jest-mock-extended';
+import set from 'lodash/set';
+import {
+	ApplicationError,
+	type INode,
+	NodeOperationError,
+	type IExecuteFunctions,
+	type INodeTypeDescription,
+} from 'n8n-workflow';
+
 import { testWorkflows, getWorkflowFilenames } from '@test/nodes/Helpers';
+import { ENABLE_LESS_STRICT_TYPE_VALIDATION } from '@utils/constants';
+
+import * as IfV2 from '../../V2/IfV2.node';
+import * as IfV2Utils from '../../V2/utils';
+
+jest.mock('lodash/set', () => jest.fn());
+
+describe('Test IF v2 Node Tests', () => {
+	afterEach(() => jest.resetAllMocks());
+
+	describe('Test IF v2 Node Workflow Tests', () => testWorkflows(getWorkflowFilenames(__dirname)));
+
+	describe('Test IF V2 Node Unit Tests', () => {
+		const baseDescriptionMock = mock<INodeTypeDescription>();
+		const node = new IfV2.IfV2(baseDescriptionMock);
+		const thisArg = mock<IExecuteFunctions>({});
+
+		it('should return false items if continue on fail is true', async () => {
+			thisArg.getInputData.mockReturnValue([{ json: {} }]);
+			thisArg.continueOnFail.mockReturnValue(true);
+
+			jest.spyOn(IfV2, 'getOptions').mockImplementation(() => {
+				throw new ApplicationError('Error');
+			});
+
+			const output = await node.execute.call(thisArg);
+			expect(output).toEqual([[], [{ json: {} }]]);
+		});
+
+		it('should throw an error if a NodeOperationError is thrown', async () => {
```

Comment:
could you set up this tests so the code throws an error instead of mocking throwing an error